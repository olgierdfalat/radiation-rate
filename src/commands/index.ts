export * from './checksum';
export * from './filterFiles';
export * from './wrongFiles';
export * from './excelExport';
