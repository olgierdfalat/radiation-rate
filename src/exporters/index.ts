export * from './worksheetsDataProvider';
export * from './excel';