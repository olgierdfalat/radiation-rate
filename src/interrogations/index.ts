export * from './interrogation';
export * from './stJude';
export * from './interrogationsFactory';
export * from './stJudeInterrogationsDataProvider';