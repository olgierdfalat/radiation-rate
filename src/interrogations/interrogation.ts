import * as fs from 'fs';
import * as models from './../models';
import { promisify } from 'util';
import yaml from 'js-yaml';

const readFile = promisify(fs.readFile);


export class Interrogation {
  private fileName: string;
  constructor(fileName: string) {
    this.fileName = fileName;
  }

  async getContent(): Promise<string> {
    return (await readFile(this.fileName)).toString();
  }

  async getData(): Promise<models.StJudeRowModel> {
    throw new Error('Not supported');
  }

  async getAsYaml() {
    return yaml.dump(await this.getData());
  }
}
