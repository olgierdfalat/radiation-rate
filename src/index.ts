import { Command } from 'commander';
import * as commands from './commands';


const command = new Command();
command
  .version('0.0.1')
  .description('CLI for converting heart valves output data into Excel format')
  .option('-c, --checksum <device>', 'generates checksums of all log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.checksum)
  .option('-e, --excluded <device>', 'generates list of excluded log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.excludedFiles)
  .option('-i, --included <device>', 'generates list of included log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.includedFiles)
  .option('-w, --invalid <device>', 'generates list of wrong log files for specific device type e.g.: StJude, Biotronik or Medtronik', commands.invalidFiles)
  .parse(process.argv);

