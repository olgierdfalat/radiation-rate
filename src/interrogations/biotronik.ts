import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class Biotronik extends Interrogation {
  async getData(): Promise<parsers.BiotronikData> {
    const content = await this.getContent();
    return new parsers.BiotronikData(content, this.filePath);
  }
}