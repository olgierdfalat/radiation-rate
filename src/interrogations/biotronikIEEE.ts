import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikIEEE extends Interrogation {
  async getData(): Promise<parsers.BiotronikIEEEData> {
    const content = await this.getContent();
    return new parsers.BiotronikIEEEData(content, this.filePath);
  }
}