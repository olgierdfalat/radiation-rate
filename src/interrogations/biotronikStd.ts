import { Interrogation } from './interrogation';
import * as parsers from '../parsers';

export class BiotronikStd extends Interrogation {
  async getData(): Promise<parsers.BiotronikStdData> {
    const content = await this.getContent();
    return new parsers.BiotronikStdData(content, this.filePath);
  }
}