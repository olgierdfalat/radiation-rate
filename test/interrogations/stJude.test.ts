import * as path from 'path';
import {StJude} from '../../src/interrogations';

describe('interrogation JtJude class', () => {
  it('should get structured data from file', async () => {
    const filePath = path.join(__dirname, '../fixtures/StJude/1000000_01.log');  
    const stJude = new StJude(filePath);
    const data = await stJude.getData();
    expect(data).toMatchSnapshot();
  });
});