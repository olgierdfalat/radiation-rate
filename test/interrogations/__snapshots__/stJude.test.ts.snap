// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interrogation StJude class should get structured data row from file 1`] = `
Array [
  Object {
    "firstValue": "Merlin",
    "id": 100,
    "name": "Programmer Marketing Name",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 3650,
    "id": 101,
    "name": "Programmer Model Number",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "11111111",
    "id": 102,
    "name": "Programmer Serial Number",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "3330 v18.2 rev 1",
    "id": 103,
    "name": "Application Software Version Number",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "IRU2.6.5_14.05.21",
    "id": 104,
    "name": "Application Build Number",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "07/07/2015 15:42:40",
    "id": 105,
    "name": "Session Timestamp",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "English (UK)",
    "id": 106,
    "name": "Programmer Language",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Fortify VR",
    "id": 200,
    "name": "Device Model Name",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "1233-40Q",
    "id": 201,
    "name": "Device Model Number",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 1111111,
    "id": 202,
    "name": "Device Serial Number",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 2015-07-09T14:42:40.000Z,
    "id": 203,
    "name": "Device Last Interrogation Date and Time",
    "secondValue": "",
    "type": "datetime",
  },
  Object {
    "firstValue": "",
    "id": 204,
    "name": "Patient ID",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 209,
    "name": "RV Lead Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "VVI",
    "id": 301,
    "name": "Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 55,
    "id": 302,
    "name": "Base Rate[bpm]",
    "secondValue": "bpm",
    "type": "numeric",
  },
  Object {
    "firstValue": "Off",
    "id": 303,
    "name": "Hysteresis Rate",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 0.5,
    "id": 305,
    "name": "RV Pulse Width[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 2.5,
    "id": 306,
    "name": "RV Pulse Amplitude[V]",
    "secondValue": "V",
    "type": "numeric",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 307,
    "name": "Ventricular Sense Configuration",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 240,
    "id": 309,
    "name": "Ventricular Pace Refractory[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "Normal",
    "id": 321,
    "name": "Magnet Response",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 353,
    "name": "RV AutoCapture",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 354,
    "name": "Rest Rate",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 370,
    "name": "Auto Intrinsic Conduction Search",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 373,
    "name": "Pre-Vent. Atrial Blanking (Pre-VAB)",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Low",
    "id": 389,
    "name": "Rate Responsive PVARP/VREF",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 215,
    "id": 390,
    "name": "Shortest PVARP/VREF[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "On",
    "id": 395,
    "name": "Suspend Notification during Rest",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Passive",
    "id": 401,
    "name": "Activity Sensor",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Auto (+0.0)",
    "id": 402,
    "name": "Activity Sensor: Threshold",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 2.1,
    "id": 403,
    "name": "Measured Average Sensor",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 8,
    "id": 404,
    "name": "Activity Sensor: Slope",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 110,
    "id": 406,
    "name": "Maximum Sensor Rate[bpm]",
    "secondValue": "bpm",
    "type": "numeric",
  },
  Object {
    "firstValue": "Fast",
    "id": 407,
    "name": "Reaction Time",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Medium",
    "id": 409,
    "name": "Activity Sensor: Recovery Time",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 3297.5,
    "id": 507,
    "name": "RV Pacing Lead Impedance[Ohm]",
    "secondValue": "Ohm",
    "type": "numeric",
  },
  Object {
    "firstValue": 7,
    "id": 533,
    "name": "Longevity Estimate[yrs]",
    "secondValue": "yrs",
    "type": "numeric",
  },
  Object {
    "firstValue": "Monitor",
    "id": 873,
    "name": "RV Lead Monitoring",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 2000,
    "id": 874,
    "name": "RV Lead Monitoring: Upper Limit",
    "secondValue": "Ohms",
    "type": "numeric",
  },
  Object {
    "firstValue": 200,
    "id": 875,
    "name": "RV Lead Monitoring: Lower Limit",
    "secondValue": "Ohms",
    "type": "numeric",
  },
  Object {
    "firstValue": "RV",
    "id": 2004,
    "name": "Ventricular Sensing Chamber",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 2008,
    "name": "RV Pulse Configuration",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Pacing Off",
    "id": 2020,
    "name": "Ventricular Noise Reversion Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 110,
    "id": 2024,
    "name": "Maximum Pacing Rate[bpm]",
    "secondValue": "bpm",
    "type": "numeric",
  },
  Object {
    "firstValue": "VVI",
    "id": 2040,
    "name": "Post-Shock Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 60,
    "id": 2041,
    "name": "Post-Shock Base Rate[bpm]",
    "secondValue": "bpm",
    "type": "numeric",
  },
  Object {
    "firstValue": 2,
    "id": 2042,
    "name": "Post-Shock Brady Pacing Pause[sec]",
    "secondValue": "sec",
    "type": "numeric",
  },
  Object {
    "firstValue": 0.5,
    "id": 2043,
    "name": "Post-Shock Duration[min]",
    "secondValue": "min",
    "type": "numeric",
  },
  Object {
    "firstValue": 7.5,
    "id": 2046,
    "name": "Post-Shock Ventricular Pulse Amplitude[V]",
    "secondValue": "V",
    "type": "numeric",
  },
  Object {
    "firstValue": 1.4,
    "id": 2047,
    "name": "Post-Shock Ventricular Pulse Width[ms]",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 3,
    "id": 2062,
    "name": "Arrhythmia Unhiding",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": "Defib with Single Tach",
    "id": 2100,
    "name": "Zone Configuration",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 280,
    "id": 2101,
    "name": "VF Detection Interval[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 350,
    "id": 2103,
    "name": "VT-1 Detection Interval[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 16,
    "id": 2105,
    "name": "VT-1 No. of Intervals",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": "Enabled",
    "id": 2106,
    "name": "Therapy Enable/Disable",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Same as VT",
    "id": 2120,
    "name": "Post VF/VT-2 Detection Interval",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 6,
    "id": 2121,
    "name": "VT Redetection",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": "Same as VT",
    "id": 2122,
    "name": "Post VT-1 Detection Interval",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 12,
    "id": 2123,
    "name": "VF No. of Intervals",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": "Ventricular Only",
    "id": 2140,
    "name": "SVT Discrimination Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Same as VF",
    "id": 2141,
    "name": "SVT Upper Limit",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "If 2 of 3",
    "id": 2142,
    "name": "Ventricular Only: VT Diagnosis Criteria",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On w/SIH",
    "id": 2145,
    "name": "Interval Stability",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 40,
    "id": 2146,
    "name": "Interval Stability Delta[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "On (60%, 7)",
    "id": 2150,
    "name": "Morphology",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On",
    "id": 2151,
    "name": "Sudden Onset",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Adaptive",
    "id": 2152,
    "name": "Sudden Onset Delta Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 20,
    "id": 2153,
    "name": "Sudden Onset Delta[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 2,
    "id": 2154,
    "name": "Interval Stability SIH Count",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": 12,
    "id": 2161,
    "name": "Interval Stability Window Size",
    "secondValue": "intervals",
    "type": "numeric",
  },
  Object {
    "firstValue": "On",
    "id": 2163,
    "name": "SVT Discrimination in VT-1",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 1,
    "id": 2180,
    "name": "Morphology Template Auto Update[day]",
    "secondValue": "day",
    "type": "numeric",
  },
  Object {
    "firstValue": "Off",
    "id": 2200,
    "name": "SVT Discrimination Timeout",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 2203,
    "name": "VT Timeout",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 125,
    "id": 2221,
    "name": "Ventricular Sense Refractory[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "Same as Defib",
    "id": 2229,
    "name": "Ventricular Pacemaker Max Sensitivity",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 0.5,
    "id": 2230,
    "name": "Ventricular Defib Max Sensitivity[mV]",
    "secondValue": "mV",
    "type": "numeric",
  },
  Object {
    "firstValue": 50,
    "id": 2231,
    "name": "Post-Sensed Ventricular Decay Delay[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 50,
    "id": 2232,
    "name": "Post-Sensed Ventricular  Threshold Start[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2233,
    "name": "Post-Paced Ventricular Decay Delay[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 1,
    "id": 2234,
    "name": "Post-Paced Ventricular Threshold Start[mV]",
    "secondValue": "mV",
    "type": "numeric",
  },
  Object {
    "firstValue": "Biphasic",
    "id": 2260,
    "name": "Waveform",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 65,
    "id": 2262,
    "name": "Defib Waveform 1st Phase Tilt[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 65,
    "id": 2264,
    "name": "Defib Waveform 2nd Phase Tilt[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": "RV to Can",
    "id": 2265,
    "name": "Shock Configuration",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Anode(+)",
    "id": 2266,
    "name": "RV Polarity",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Tilt",
    "id": 2267,
    "name": "HV Output Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 3,
    "id": 2291,
    "name": "VT-1 Therapy 1 No. Bursts",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 85,
    "id": 2292,
    "name": "VT-1 Therapy 1 BCL[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 8,
    "id": 2293,
    "name": "VT-1 Therapy 1 No. Stimuli",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "Off",
    "id": 2294,
    "name": "VT-1 Therapy 1 Extrastimuli Per Burst",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 2295,
    "name": "VT-1 Therapy 1 Readaptive",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 200,
    "id": 2296,
    "name": "VT-1 Therapy 1 Min. BCL[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "On (Dec)",
    "id": 2297,
    "name": "VT-1 Therapy 1 Scan Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 10,
    "id": 2298,
    "name": "VT-1 Therapy 1 Scan Step Size[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": "Off",
    "id": 2300,
    "name": "VT-1 Therapy 1 Ramp",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 7.5,
    "id": 2302,
    "name": "ATP Pulse Amplitude[V]",
    "secondValue": "V",
    "type": "numeric",
  },
  Object {
    "firstValue": 1.1,
    "id": 2303,
    "name": "ATP Pulse Width[ms]",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "Adaptive",
    "id": 2304,
    "name": "VT-1 Therapy 1 BCL Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "ATP",
    "id": 2320,
    "name": "VT-1 Therapy 1 Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "CVRT",
    "id": 2321,
    "name": "VT-1 Therapy 2 Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "CVRT",
    "id": 2322,
    "name": "VT-1 Therapy 3 Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "CVRT x 2",
    "id": 2323,
    "name": "VT-1 Therapy 4 Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 25,
    "id": 2327,
    "name": "VT-1 Energy 2[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": 35,
    "id": 2329,
    "name": "VT-1 Energy 3[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": 20,
    "id": 2331,
    "name": "VT-1 Energy 4[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": "Same as Defib",
    "id": 2368,
    "name": "CVRT Waveform 1st Phase Tilt",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Same as Defib",
    "id": 2369,
    "name": "CVRT Waveform 2nd Phase Tilt",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 31,
    "id": 2382,
    "name": "VF Energy 1[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": 40,
    "id": 2384,
    "name": "VF Energy 2[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": 40,
    "id": 2386,
    "name": "VF Energy 3[J]",
    "secondValue": "J",
    "type": "numeric",
  },
  Object {
    "firstValue": "ATP While Charging",
    "id": 2387,
    "name": "VF Therapy 1 Type",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 240,
    "id": 2388,
    "name": "Max VF Interval for ATP[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 85,
    "id": 2389,
    "name": "VF Therapy BCL[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": "Adaptive",
    "id": 2390,
    "name": "VF Therapy BCL Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 170,
    "id": 2391,
    "name": "VF Therapy Min. BCL[ms]",
    "secondValue": "ms",
    "type": "numeric",
  },
  Object {
    "firstValue": 1,
    "id": 2392,
    "name": "VF Therapy No. Bursts",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 8,
    "id": 2393,
    "name": "VF Therapy No. Stimuli",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "Off",
    "id": 2394,
    "name": "VF Therapy Ramp",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "V Sense Amp",
    "id": 2400,
    "name": "Episode Settings: EGM 1 Configuration",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Can",
    "id": 2404,
    "name": "Episode Settings: Custom EGM 1 Anode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "V tip",
    "id": 2405,
    "name": "Episode Settings: Custom EGM 1 Cathode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 14,
    "id": 2407,
    "name": "VT/VF Pre-Trigger Max Duration[sec]",
    "secondValue": "sec",
    "type": "numeric",
  },
  Object {
    "firstValue": 1,
    "id": 2408,
    "name": "VT/VF EGM Max Duration[min]",
    "secondValue": "min",
    "type": "numeric",
  },
  Object {
    "firstValue": "ANONYMOUS NAME",
    "id": 2430,
    "name": "Patient Name",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "",
    "id": 2432,
    "name": "Follow-up Physician",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 20,
    "id": 2440,
    "name": "Ejection Fraction[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": "Atrial Fibrillation (AF), AV Block, 3rd degree, Dilated Cardiomyopathy (CHF/DCM), Non-Ischemic Cardiomyopathy",
    "id": 2441,
    "name": "Indications for Implant: List",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 2012-07-06T23:00:00.000Z,
    "id": 2442,
    "name": "Implant Date: Device",
    "secondValue": "",
    "type": "datetime",
  },
  Object {
    "firstValue": "<not set>",
    "id": 2448,
    "name": "Manufacturer: RV Lead",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "",
    "id": 2450,
    "name": "Model Number: Other RV Lead",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "",
    "id": 2469,
    "name": "RV Lead Serial Number",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On",
    "id": 2480,
    "name": "Low Battery Notification Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On",
    "id": 2481,
    "name": "HV Charge Notification Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On",
    "id": 2482,
    "name": "Firmware ROM/RAM Reset Notification Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "On",
    "id": 2483,
    "name": "SOSD Notification Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Off",
    "id": 2485,
    "name": "Ventricular PLI Notification Mode",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 7,
    "id": 2487,
    "name": "Stimulus Duration[sec]",
    "secondValue": "sec",
    "type": "numeric",
  },
  Object {
    "firstValue": 10,
    "id": 2488,
    "name": "Re-Notification Frequency[hours]",
    "secondValue": "hours",
    "type": "numeric",
  },
  Object {
    "firstValue": 45,
    "id": 2489,
    "name": "Number of Notifications",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2630,
    "name": "Total Number of Episodes",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2631,
    "name": "Number of Fib Episodes with Therapy Aborted",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2632,
    "name": "Number of Fib Episodes with Therapy Delivered",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2633,
    "name": "Number Of Tach B Episodes With All Diagnosis Inhibited",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2636,
    "name": "Number Of Tach A Episodes With All Diagnoses Inhibited",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2639,
    "name": "Number Of MTD/MTT Episodes With Diagnosis Inhibited By BA",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2660,
    "name": "Delivered Shocks For Episode Therapies",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2661,
    "name": "Tach A Episodes Where ATP Delivered",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2662,
    "name": "Tach B Episodes Where ATP Delivered",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2670,
    "name": "Total Number of Aborted Shocks",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 90,
    "id": 2681,
    "name": "Event Histogram Percent Paced In Ventricle[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2700,
    "name": "Device Charging History Level 1",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2701,
    "name": "Device Charging History Level 2",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2702,
    "name": "Device Charging History Level 3",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2703,
    "name": "Device Charging History Level 4",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2704,
    "name": "Device Charging History Level 5",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2705,
    "name": "Device Charging History Level 6",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": "",
    "id": 2706,
    "name": "Device Charging History Level 75",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 25,
    "id": 2707,
    "name": "Device Charging History Level 8",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 71,
    "id": 2709,
    "name": "Ventricular Paced - Lifetime (RVP)[%]",
    "secondValue": "%",
    "type": "numeric",
  },
  Object {
    "firstValue": 11.7,
    "id": 2722,
    "name": "Ventricular Signal Amplitude[mV]",
    "secondValue": "mV",
    "type": "numeric",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 2724,
    "name": "Test Polarity of Ventricular Signal Amplitude",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": 285.75,
    "id": 2730,
    "name": "HV Lead Impedance[Ohm]",
    "secondValue": "Ohm",
    "type": "numeric",
  },
  Object {
    "firstValue": 8.4,
    "id": 2740,
    "name": "Last Cap Maintenance Charge Duration[sec]",
    "secondValue": "sec",
    "type": "numeric",
  },
  Object {
    "firstValue": 845,
    "id": 2741,
    "name": "Last Cap Maintenance Voltage[V]",
    "secondValue": "V",
    "type": "numeric",
  },
  Object {
    "firstValue": 4,
    "id": 2742,
    "name": "Charge Interval[months]",
    "secondValue": "months",
    "type": "numeric",
  },
  Object {
    "firstValue": 2015-06-07T14:54:15.000Z,
    "id": 2743,
    "name": "Last Max Charge Date",
    "secondValue": "",
    "type": "datetime",
  },
  Object {
    "firstValue": 8.275,
    "id": 2745,
    "name": "Last HV Cap Charge Duration[sec]",
    "secondValue": "sec",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2754,
    "name": "Total Number of AT/AF Episodes Since Last Cleared",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 0,
    "id": 2755,
    "name": "Total Time in AT/AF Recent Week",
    "secondValue": "",
    "type": "numeric",
  },
  Object {
    "firstValue": 2015-10-30T00:00:00.000Z,
    "id": 2765,
    "name": "Right Ventricular Pacing Lead Impedance Notification Time Stamp (Date)",
    "secondValue": "",
    "type": "datetime",
  },
  Object {
    "firstValue": "20:23:48",
    "id": 2766,
    "name": "Right Ventricular Pacing Lead Impedance Notification Time Stamp (Time)",
    "secondValue": "",
    "type": "time",
  },
  Object {
    "firstValue": false,
    "id": 2902,
    "name": "Committed",
    "secondValue": "",
    "type": "boolean",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 2905,
    "name": "V Sense Polarity",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Bipolar",
    "id": 2907,
    "name": "V Pace Polarity",
    "secondValue": "",
    "type": "string",
  },
  Object {
    "firstValue": "Decrement",
    "id": 2911,
    "name": "VT-1 Therapy 1 Sequence",
    "secondValue": "",
    "type": "string",
  },
]
`;
